ads/google/a4a/performance.js

```diff
@@ -188,42 +158,59 @@ export class GoogleAdLifecycleReporter extends BaseLifecycleReporter {
+   * names or values with the ID of the ad slot on the page.
+   *
    * @param {string} name  Stage name to ping out.  Should be one of the ones
    * from `LIFECYCLE_STAGES`.  If it's an unknown name, it will still be pinged,
    * but the stage ID will be set to `9999`.
    * @override
    */
   sendPing(name) {
-    this.emitPing_(this.buildPingAddress_(name));
+    const url = this.buildPingAddress_(name, this.extraVariables_);
+    if (url) {
+      this.emitPing_(url);
+    }
   }
 
   /**
    * @param {string} name  Metric name to send.
+   * @param {!Object<string, string>=} opt_extraParams
    * @returns {string}  URL to send metrics to.
    * @private
    */
-  buildPingAddress_(name) {
+  buildPingAddress_(name, opt_extraParams) {
     const stageId = LIFECYCLE_STAGES[name] || 9999;
-    const delta = Date.now() - this.initTime_;
-    // Note: QQid comes from a network header and eid could, potentially, be
-    // injected by a publisher.  Treat both of them as unverified user content
-    // and encode before inserting them into URI.
-    const encodedQqid = this.qqid_ ?
-        encodeURIComponent(this.qqid_) : false;
-    const qqidParam = encodedQqid ?
-        `&qqid.${this.slotId_}=${encodedQqid}` : '';
-    const eid = this.element_.getAttribute(EXPERIMENT_ATTRIBUTE);
-    const eidParam = eid ? `&e=${encodeURIComponent(eid)}` : '';
-    const pingUrl = `${this.pingbackAddress_}?` +
-        `s=${this.namespace_}` +
-        `&v=2&it=${name}.${delta},${name}_${this.slotId_}.${delta}` +
-        `&rt=stage.${stageId},slotId.${this.slotId_}` +
-        `&c=${this.correlator_}` +
-        '&rls=$internalRuntimeVersion$' +
-        `${eidParam}${qqidParam}` +
-        `&it.${this.slotName_}=${name}.${delta}` +
-        `&rt.${this.slotName_}=stage.${stageId}` +
-        `&met.${this.slotName_}=stage_${stageId}.${delta}`;
+    const delta = Math.round(this.getDeltaTime_());
+    let extraParams = '';
+    if (opt_extraParams) {
+      extraParams = serializeQueryString(opt_extraParams);
+      if (extraParams != '') {
+        // Note: Using sync URL replacer here, rather than async, for a number
+        // of reasons:
+        //   - Don't want to block pings waiting for potentially delayed bits
+        //     of information.
+        //   - Don't (currently) need access to any properties that are
+        //     available async only.
+        //   - Don't want to pass through expandStringAsync if there are zero
+        //     extra params, but async would force us to (or to maintain two
+        //     code branches).
+        // TODO(ampproject/a4a): Change to async if/when there's a need to
+        // expand async-only parameters.  E.g., we'd like to have scroll_y
+        // offset, but it's not currently available through url-replacement.
+        // If it becomes available, it's likely to be an async parameter.
+        extraParams = this.urlReplacer_./*OK*/expandStringSync(extraParams, {
```

Comment:
Thought about it.  Seemed easier to detect the case that there were no params, so no ping should be issued, this way.